internal/infrastructure/kubernetes/infra.go

```diff
@@ -0,0 +1,88 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/envoyproxy/gateway/internal/envoygateway/config"
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+const (
+	KindServiceAccount Kind = "ServiceAccount"
+)
+
+type Kind string
+
+// Infra holds all the translated Infra IR resources and provides
+// the scaffolding for the managing Kubernetes infrastructure.
+type Infra struct {
+	Client    client.Client
+	Log       logr.Logger
```

Comment:
The logger shouldn't be stored in a struct, on the principle that it is kin with a Context; the [`context` docs](https://pkg.go.dev/context) tell us:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

I'd even argue that we shouldn't be passing around a `logr.Logger`, we should be passing around a `context.Context` that we call `logr.FromContext(ctx)` on at the leaf of where it's needed.

An exception has to be made for Nick's log adapter, since it has to go-control-plane doesn't pass the callback functions a context or logger, and that's the interface we need to implement.  We have no such constraint here.